src/test/java/PageFactory/MainPage.java

```diff
@@ -0,0 +1,48 @@
+package PageFactory;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class MainPage extends BaseStepPage {
+
+    @FindBy(css = "a.button2_theme_mail-white")
+    private WebElement generalButton;
+
+    @FindBy(id = "passp-field-login")
+    private WebElement loginField;
+
+    @FindBy(id = "passp:sign-in")
+    private WebElement signIn;
+
+    @FindBy(id = "passp-field-passwd")
+    private WebElement passwordField;
+    @FindBy(css = "button#passp\\:sign-in")
+    private WebElement submitButton;
+
+
+    @FindBy(css = "a.user-account_left-name>div>img")
+    private WebElement userNameLinkLogedIn;
+
+    @FindBy(xpath = "//span[@class='user-account__name']/following-sibling::div")
+    private WebElement userNameText;
+
+    public WebElement getUserNameLinkText() {
+        return userNameText;
+    }
```

Comment:
The sense of page object is to encapsulate actions with elements. If you have public method, which returns WebElement, then you violate pattern implementation. Remove this method, create method `public String getUsername()`